### ISSUE TITLE
account_lines paginating when there is only one "line" (Version: 1.9.4)

### ISSUE THREAD

**Issue by zgrguric** [XRPLF/rippled#4340]
## Issue Description
`account_lines` command of account `rBLadExFZKY7AqpTDxqSoWJgmgVZyA6wcX` returns marker where there should be no markers.

Also with limit of 100 there is no "lines" returned, when querying 200 there is 1 item in "lines" returned.

Note: There are lot of NFTs in this account.

## Steps to Reproduce
Use https://xrpl.org/websocket-api-tool.html#account_lines

```
{
  "id": 2,
  "command": "account_lines",
  "account": "rBLadExFZKY7AqpTDxqSoWJgmgVZyA6wcX",
  "ledger_index": "validated",
  "limit":100
}
```
This above returns marker and lines response is empty.

```
{
  "id": 2,
  "command": "account_lines",
  "account": "rBLadExFZKY7AqpTDxqSoWJgmgVZyA6wcX",
  "ledger_index": "validated",
  "limit":200
}
```
This above returns marker and lines response returns one (1) item in "lines".  
Using marker returns "invalidParams" error.

Results are consistant for xrplcluster.com, s1.ripple.com and s2.ripple.com.

## Expected Result
No marker, only one trustline returned with no limit defined or limit 1 minimum,

## Actual Result

As you can see "marker" value filled and "lines" is empty.
```
{
  "id": 2,
  "result": {
    "account": "rBLadExFZKY7AqpTDxqSoWJgmgVZyA6wcX",
    "ledger_hash": "772BB5FB46D0E476E51D03AC8BF2F8992D0461910806FC71545AF63F4A47FE92",
    "ledger_index": 75654696,
    "limit": 100,
    "lines": [],
    "marker": "36E86F954365F2723351DB927416B40330D47E064FCE0365FCB6A09C5E5A80F1,4",
    "validated": true,
    "warnings": [
      {
        "id": 1004,
        "message": "This is a reporting server.  The default behavior of a reporting server is to only return validated data. If you are looking for not yet validated data, include \"ledger_index : current\" in your request, which will cause this server to forward the request to a p2p node. If the forward is successful the response will include \"forwarded\" : \"true\""
      }
    ]
  },
  "status": "success",
  "type": "response"
}
```

## Supporting Files
Discovered by KegsRP https://twitter.com/KegsRp/status/1590593708130697216

**Comment by ximinez** [comment]
I've been able to reproduce this, and should have a fix PR soon.

**Comment by ximinez** [comment]
#4361 resolves the issue with the `marker` being invalid. It is expected that fewer than `limit` lines may be returned along with a `marker`, because of an optimization released in 1.8.2 to reduce load on rippled servers by counting _all_ objects up to the limit instead of just the requested object type.

Please read the description of that PR for more details.
